e2e/appActions.js

```diff
@@ -101,8 +101,19 @@ async function createDomainObjectWithDefaults(
   // If there are any further parameters, fill them in
   for (const [key, value] of Object.entries(customParameters)) {
     const input = page.locator(`form[name="mctForm"] ${key}`);
-    await input.fill('');
-    await input.fill(value);
+    const inputType = await input.getAttribute('type');
+
+    if (inputType === 'checkbox') {
+      // Locate the slider span associated with the checkbox
+      const slider = input.locator('+ .c-toggle-switch__slider');
+      const isChecked = await input.isChecked();
+      if ((value === 'true' && !isChecked) || (value === 'false' && isChecked)) {
+        await slider.click();
+      }
+    } else {
+      await input.fill('');
+      await input.fill(value);
+    }
```

Comment:
we ought to move away from this pattern in favor of improving a11y for the different components in a form. i can see this function growing in size and complexity and becoming less and less maintainable as more tests depend on it

Reply:
maybe we can sync and find a solution